### PR TITLE
Add support for getting vectors by ids (#23450)

### PR DESCRIPTION
**Pull request by bigsheeper**
/kind enhancement

issue: https://github.com/milvus-io/milvus/issues/23449

### CODE REVIEW

File: internal/core/src/index/VectorMemIndex.cpp [hunk 1]
```diff
@@ -222,4 +222,29 @@ VectorMemIndex::parse_config(Config& config) {
     CheckParameter<int>(config, knowhere::indexparam::SEARCH_K, stoi_closure, std::nullopt);
 }
 
+const bool
+VectorMemIndex::HasRawData() const {
+    return index_->HasRawData(GetMetricType());
+}
+
+const std::vector<uint8_t>
```
Comment (czs007) [inline internal/core/src/index/VectorMemIndex.cpp:230]: remove const qualifier to enable move semantics

File: internal/proxy/task_search.go [hunk 2]
```diff
@@ -551,6 +587,99 @@ func (t *searchTask) searchShard(ctx context.Context, nodeID int64, qn types.Que
...
+	//       \/
+	//  3  2  5  4  1  (result ids)
+	// v3 v2 v5 v4 v1  (result vectors)
+	// ===========================================
+	pkFieldData, err := typeutil.GetPrimaryFieldData(queryResult.GetFieldsData(), pkField)
+	if err != nil {
+		return err
+	}
+	offsets := make(map[any]int)
+	for i := 0; i < typeutil.GetDataSize(pkFieldData); i++ {
```
Comment (czs007) [inline internal/proxy/task_search.go:660]: Just need to implement a function to retrieve the length of the PKField.

File: internal/util/typeutil/schema_test.go [hunk 3]
```diff
@@ -529,6 +529,21 @@ func genFieldData(fieldName string, fieldID int64, fieldType schemapb.DataType,
 			},
 			FieldId: fieldID,
 		}
+	case schemapb.DataType_String:
```
Comment (czs007) [inline internal/util/typeutil/schema_test.go:532]: remove this

File: internal/util/typeutil/schema.go [hunk 4]
```diff
@@ -814,6 +814,30 @@ func GetSizeOfIDs(data *schemapb.IDs) int {
...
+		return len(fieldData.GetScalars().GetBoolData().GetData())
+	case schemapb.DataType_Int8, schemapb.DataType_Int16, schemapb.DataType_Int32:
+		return len(fieldData.GetScalars().GetIntData().GetData())
+	case schemapb.DataType_Int64:
+		return len(fieldData.GetScalars().GetLongData().GetData())
+	case schemapb.DataType_Float:
+		return len(fieldData.GetScalars().GetFloatData().GetData())
+	case schemapb.DataType_Double:
+		return len(fieldData.GetScalars().GetDoubleData().GetData())
+	case schemapb.DataType_String:
```
Comment (czs007) [inline internal/util/typeutil/schema.go:829]: remove this

**Comment by bigsheeper** [comment]
run ut

**Comment by bigsheeper** [comment]
rerun ut

**Comment by bigsheeper** [comment]
rerun ut

**Comment by bigsheeper** [comment]
rerun ut

**Pull request merged**